### PR TITLE
Remove pyarrow dependence

### PR DESCRIPTION
**Pull request by zenghsh3**
- use cloudpickle as defaut serializing/deserializing
- add performance comparison

### CODE REVIEW

**Comment by zenghsh3** [comment]
Speed test results: (last test result: https://github.com/PaddlePaddle/PARL/pull/77#issuecomment-491262491)
Result in 12 CPU machine (Intel(R) Xeon(R) Gold 6128 CPU @ 3.40GHz), Ubuntu:

- Python 3.5.2

Case 0, Average dump and load argument time: 15.33392698764801
Case 1, Average dump and load argument time: 0.0045697689056396484
Case 2, Average dump and load argument time: 2.720403242111206

Case 0, Average dump and load return time: 15.108179903030395
Case 1, Average dump and load return time: 0.004416608810424804
Case 2, Average dump and load return time: 2.7147656440734864

- Python 3.8.5

Case 0, Average dump and load argument time: 1.3110248804092408
Case 1, Average dump and load argument time: 0.007070708274841309
Case 2, Average dump and load argument time: 1.0784552097320557

Case 0, Average dump and load return time: 1.266370987892151
Case 1, Average dump and load return time: 0.007046055793762207
Case 2, Average dump and load return time: 1.0695510149002074

File: docs/zh_CN/xparl/serialize.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+这里提供了pyarrow和cloudpickle在不同数据下的序列化和反序列化的平均耗时作为参考：
+
+- 测试数据一：`data = [np.random.RandomState(0).randn(50, 50)] * 10`
+- 测试数据二：`data = [np.random.RandomState(0).randn(500, 500)] * 10`
+- 测试数据三：`data = [np.random.RandomState(0).randn(5000, 5000)] * 10`
+- 测试数据四：`data = np.random.RandomState(0).randn(5000, 50000)`
+
+> pyarrow版本：python2使用pyarrow==0.16.0，python3中使用pyarrow==2.0.0
+
+<img src="./.images/dataset1.png" width="400"/> <img src="./.images/dataset2.png" width="400"/>
```
Comment (TomorrowIsAnOtherDay) [inline docs/zh_CN/xparl/serialize.md:17]: The title of images should be written in English. We have to rewrite the English documentation in the near term.

File: docs/zh_CN/xparl/serialize.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+<img src="./.images/dataset3.png" width="400"/> <img src="./.images/dataset4.png" width="400"/>
+
+### 对比结论
+- 在序列化/反序列化**超大Numpy矩阵**时，pyarrow表现明显比cloudpickle好
+- 使用高版本的Python也能提升序列化性能。（主要是python3.8+版本对pickle进行了升级，支持protocol=5）
```
Comment (TomorrowIsAnOtherDay) [inline docs/zh_CN/xparl/serialize.md]: 高版本的Python -> 3.8+版本的Python

File: setup.py [hunk 3]
```diff
@@ -72,10 +72,11 @@ def find_version(*file_paths):
     install_requires=[
         "termcolor>=1.1.0",
         "pyzmq==18.0.1",
-        'pyarrow==0.16.0; python_version<"3"',
-        'pyarrow==0.17.1; python_version>="3"',
+        #'pyarrow==0.16.0; python_version<"3"',
```
Comment (TomorrowIsAnOtherDay) [inline setup.py]: Please remove the commented lines.

File: docs/zh_CN/xparl/serialize.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# **序列化/反序列化加速**
```
Comment (TomorrowIsAnOtherDay) [inline docs/zh_CN/xparl/serialize.md]: 序列化/反序列化加速（非必须）

**Comment by zenghsh3** [comment]
Test throughput with IMPALA/A2C algorithm in a GeForce GTX 1080 Ti GPU machine with 12 CPUs.


- IMPALA (Sample steps in 10 minutes and 12 actors):

| Version     | Sample steps (Python 3.5.2, paddlepaddle-gpu==1.6.1) | Sample steps (Python 3.8.5, paddlepaddle-gpu==1.8.5) |
|-------------|--------------|--------------|
| parl==1.3.3 |     2258750 |  2265750   |
| This PR     |  2227500    |   2216500    |

- A2C (Sample steps in 10 minutes and 5 actors):

| Version     | Sample steps (Python 3.5.2, paddlepaddle-gpu==1.6.1) | Sample steps (Python 3.8.5, paddlepaddle-gpu==1.8.5) |
|-------------|--------------|--------------|
| parl==1.3.3 |  753500    |   761000  |
| This PR     | 483500     |  770500     |


> There is no big difference in IMPALA, but a big difference in A2C and lower version of python(< python 3.8).

File: docs/zh_CN/Overview.md [hunk 5]
```diff
@@ -78,6 +78,7 @@
             <li><a href="xparl/example.md">加速案例</a></li>
             <li><a href="xparl/debug.md">如何debug</a></li>
             <li><a href="xparl/distribute_files.md">分发本地文件</a></li>
+            <li><a href="xparl/serialize.md">序列化加速</a></li>
```
Comment (TomorrowIsAnOtherDay) [inline docs/zh_CN/Overview.md]: 序列化加速（非必须）

File: docs/zh_CN/xparl/serialize.md (hunk 1 above)
Comment (zenghsh3) [reply docs/zh_CN/xparl/serialize.md:17]: done

File: docs/zh_CN/xparl/serialize.md (hunk 2 above)
Comment (zenghsh3) [reply docs/zh_CN/xparl/serialize.md]: done

File: setup.py (hunk 3 above)
Comment (zenghsh3) [reply setup.py]: done

File: docs/zh_CN/xparl/serialize.md (hunk 4 above)
Comment (zenghsh3) [reply docs/zh_CN/xparl/serialize.md]: done

File: docs/zh_CN/Overview.md (hunk 5 above)
Comment (zenghsh3) [reply docs/zh_CN/Overview.md]: done

**Review by TomorrowIsAnOtherDay — APPROVED** [review]
Looks good to me.

**Pull request merged**